### DISCUSSION TITLE
does django-select2 work with django rest framework?

### DISCUSSION THREAD

**Discussion by chrcorn** [codingjoe/django-select2#90] [Q&A] [answered]
is django-select2 functional with django rest framework?

since drf uses serializers and not forms, widgets cannot be defined

**Comment by codingjoe** [comment, marked as answer]
Hi @chrcorn,

Thank you for reaching out. I'll try to answer your question the best I can, making some assumptions, please let me know if I misunderstood your question.

So, this packages tries to deliver an out-of-the-box solution for Django's forms. However, some use cases require special integrations. If you don't use Django's forms, this package may not be suited for you. For you use case, I recommend building a custom integration of the `Select2` library. You can build the endpoint, that delivers you the select options via DRF. If you have a very sophisticated frontend, you may even go ahead and write the frontend bit yourself in your JS framework.

Whatever path you choose, I hope this package serves you as a guide or documentation on how the frontend and backend can work together and what security implications a custom integration might have.

I hope this answer helps yo a little.

Best,
Joe

**Reply by chrcorn** [reply]
thank you for the reply!
